### ISSUE TITLE
The module destroys the stable diffusion webui environment

### ISSUE THREAD

**Issue by schoenid** [WASasquatch/was-node-suite-comfyui#457]
Installed with the ComfyUI manager.
After a restart of the webui, it's no more working and I've got the following problem:

```
venv "G:\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: v1.10.1
Commit hash: 82a973c04367123ae98bd9abdf80d9eda9b910e2
Installing sd-webui-comfyui requirement: changing portalocker version from 2.10.1 to 2.7.0
Launching Web UI with arguments: --xformers --medvram-sdxl --opt-split-attention-v1 --theme dark --api
Traceback (most recent call last):
  File "G:\stable-diffusion-webui\launch.py", line 48, in <module>
    main()
  File "G:\stable-diffusion-webui\launch.py", line 44, in main
    start()
  File "G:\stable-diffusion-webui\modules\launch_utils.py", line 465, in start
    import webui
  File "G:\stable-diffusion-webui\webui.py", line 13, in <module>
    initialize.imports()
  File "G:\stable-diffusion-webui\modules\initialize.py", line 39, in imports
    from modules import processing, gradio_extensons, ui  # noqa: F401
  File "G:\stable-diffusion-webui\modules\processing.py", line 31, in <module>
    from ldm.data.util import AddMiDaS
  File "G:\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\data\util.py", line 3, in <module>
    from ldm.modules.midas.api import load_midas_transform
  File "G:\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\midas\api.py", line 11, in <module>
    from ldm.modules.midas.midas.transforms import Resize, NormalizeImage, PrepareForNet
  File "G:\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\midas\midas\transforms.py", line 6, in <module>
    def apply_min_size(sample, size, image_interpolation_method=cv2.INTER_AREA):
AttributeError: module 'cv2' has no attribute 'INTER_AREA'
Drücken Sie eine beliebige Taste . . .
```

A manual delete of the module from G:\stable-diffusion-webui\extensions\sd-webui-comfyui\ComfyUI\custom_nodes does not help.
It is required to delete and rebuild venv (G:\stable-diffusion-webui\venv) to recover the webui.

Gave it a second try, loaded ComfyUI and installed the module, the webui crashed again.

![image](https://github.com/user-attachments/assets/8a1c4115-496f-469f-8b67-069761849fab)

version: [v1.10.1](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/82a973c04367123ae98bd9abdf80d9eda9b910e2)  •  python: 3.10.11  •  torch: 2.1.2+cu121  •  xformers: 0.0.23.post1  •  gradio: 3.41.2  •  checkpoint: [879db523c3](https://google.com/search?q=879db523c30d3b9017143d56705015e15a2cb5628762c11d086fed9538abd7fd)

System:	Windows 10 Pro
Version	22H2
Betriebssystembuild	19045.4651

**Comment by schoenid** [comment]
Tried a manual installation (recommended).
Seems to work but has a few  problems at the end:

```
Microsoft Windows [Version 10.0.19045.4651]
(c) Microsoft Corporation. Alle Rechte vorbehalten.

G:\stable-diffusion-webui\extensions\sd-webui-comfyui\ComfyUI\custom_nodes\was-node-suite-comfyui>pip install -r requirements.txt
Collecting git+https://github.com/WASasquatch/img2texture.git (from -r requirements.txt (line 3))
  Cloning https://github.com/WASasquatch/img2texture.git to c:\users\danie\appdata\local\temp\pip-req-build-wpaw1d4b
  Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/img2texture.git 'C:\Users\danie\AppData\Local\Temp\pip-req-build-wpaw1d4b'
  Resolved https://github.com/WASasquatch/img2texture.git to commit a546609a915caf1dcb84a15bf44b5b639ad924e6
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting git+https://github.com/WASasquatch/cstr (from -r requirements.txt (line 4))
  Cloning https://github.com/WASasquatch/cstr to c:\users\danie\appdata\local\temp\pip-req-build-nik62q83
  Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/cstr 'C:\Users\danie\AppData\Local\Temp\pip-req-build-nik62q83'
  Resolved https://github.com/WASasquatch/cstr to commit 0520c29a18a7a869a6e5983861d6f7a4c86f8e9b
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting git+https://github.com/WASasquatch/ffmpy.git (from -r requirements.txt (line 13))
  Cloning https://github.com/WASasquatch/ffmpy.git to c:\users\danie\appdata\local\temp\pip-req-build-cfwkmpz6
  Running command git clone --filter=blob:none --quiet https://github.com/WASasquatch/ffmpy.git 'C:\Users\danie\AppData\Local\Temp\pip-req-build-cfwkmpz6'
  Resolved https://github.com/WASasquatch/ffmpy.git to commit f000737698b387ffaeab7cd871b0e9185811230d
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting cmake (from -r requirements.txt (line 1))
  Using cached cmake-3.30.2-py3-none-win_amd64.whl.metadata (6.1 kB)
Collecting fairscale>=0.4.4 (from -r requirements.txt (line 2))
  Using cached fairscale-0.4.13-py3-none-any.whl
Requirement already satisfied: gitpython in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 5)) (3.1.43)
Requirement already satisfied: imageio in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 6)) (2.34.2)
Requirement already satisfied: joblib in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 7)) (1.4.2)
Requirement already satisfied: matplotlib in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 8)) (3.9.0)
Requirement already satisfied: numba in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 9)) (0.60.0)
Requirement already satisfied: numpy in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 10)) (1.26.4)
Collecting opencv-python-headless<=4.7.0.72 (from opencv-python-headless[ffmpeg]<=4.7.0.72->-r requirements.txt (line 11))
  Using cached opencv_python_headless-4.7.0.72-cp37-abi3-win_amd64.whl.metadata (18 kB)
Collecting pilgram (from -r requirements.txt (line 12))
  Using cached pilgram-1.2.1-py3-none-any.whl.metadata (4.0 kB)
Collecting rembg (from -r requirements.txt (line 14))
  Using cached rembg-2.0.58-py3-none-any.whl.metadata (16 kB)
Requirement already satisfied: scikit-image>=0.20.0 in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 15)) (0.24.0)
Requirement already satisfied: scikit-learn in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 16)) (1.5.0)
Requirement already satisfied: scipy in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 17)) (1.14.0)
Requirement already satisfied: timm>=0.4.12 in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 18)) (1.0.7)
Requirement already satisfied: tqdm in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 19)) (4.66.4)
Requirement already satisfied: transformers in e:\programme\python\python310\lib\site-packages (from -r requirements.txt (line 20)) (4.30.2)
Requirement already satisfied: torch>=1.8.0 in e:\programme\python\python310\lib\site-packages (from fairscale>=0.4.4->-r requirements.txt (line 2)) (2.1.2+cu121)
Collecting pillow<10,>=9.2 (from img2texture==1.0.6->-r requirements.txt (line 3))
  Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl.metadata (9.7 kB)
Requirement already satisfied: gitdb<5,>=4.0.1 in e:\programme\python\python310\lib\site-packages (from gitpython->-r requirements.txt (line 5)) (4.0.11)
Requirement already satisfied: contourpy>=1.0.1 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (1.2.1)
Requirement already satisfied: cycler>=0.10 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (4.53.0)
Requirement already satisfied: kiwisolver>=1.3.1 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (1.4.5)
Requirement already satisfied: packaging>=20.0 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (24.1)
Requirement already satisfied: pyparsing>=2.3.1 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (3.1.2)
Requirement already satisfied: python-dateutil>=2.7 in e:\programme\python\python310\lib\site-packages (from matplotlib->-r requirements.txt (line 8)) (2.9.0.post0)
Requirement already satisfied: llvmlite<0.44,>=0.43.0dev0 in e:\programme\python\python310\lib\site-packages (from numba->-r requirements.txt (line 9)) (0.43.0)
Requirement already satisfied: jsonschema in e:\programme\python\python310\lib\site-packages (from rembg->-r requirements.txt (line 14)) (4.22.0)
Requirement already satisfied: onnxruntime in e:\programme\python\python310\lib\site-packages (from rembg->-r requirements.txt (line 14)) (1.15.0)
Collecting pooch (from rembg->-r requirements.txt (line 14))
  Using cached pooch-1.8.2-py3-none-any.whl.metadata (10 kB)
Collecting pymatting (from rembg->-r requirements.txt (line 14))
  Using cached PyMatting-1.1.12-py3-none-any.whl.metadata (7.4 kB)
Requirement already satisfied: networkx>=2.8 in e:\programme\python\python310\lib\site-packages (from scikit-image>=0.20.0->-r requirements.txt (line 15)) (3.3)
Requirement already satisfied: tifffile>=2022.8.12 in e:\programme\python\python310\lib\site-packages (from scikit-image>=0.20.0->-r requirements.txt (line 15)) (2024.6.18)
Requirement already satisfied: lazy-loader>=0.4 in e:\programme\python\python310\lib\site-packages (from scikit-image>=0.20.0->-r requirements.txt (line 15)) (0.4)
Requirement already satisfied: threadpoolctl>=3.1.0 in e:\programme\python\python310\lib\site-packages (from scikit-learn->-r requirements.txt (line 16)) (3.5.0)
Requirement already satisfied: torchvision in e:\programme\python\python310\lib\site-packages (from timm>=0.4.12->-r requirements.txt (line 18)) (0.16.2+cu121)
Requirement already satisfied: pyyaml in e:\programme\python\python310\lib\site-packages (from timm>=0.4.12->-r requirements.txt (line 18)) (6.0.1)
Requirement already satisfied: huggingface_hub in e:\programme\python\python310\lib\site-packages (from timm>=0.4.12->-r requirements.txt (line 18)) (0.23.4)
Requirement already satisfied: safetensors in e:\programme\python\python310\lib\site-packages (from timm>=0.4.12->-r requirements.txt (line 18)) (0.4.3)
Requirement already satisfied: colorama in e:\programme\python\python310\lib\site-packages (from tqdm->-r requirements.txt (line 19)) (0.4.6)
Requirement already satisfied: filelock in e:\programme\python\python310\lib\site-packages (from transformers->-r requirements.txt (line 20)) (3.15.4)
Requirement already satisfied: regex!=2019.12.17 in e:\programme\python\python310\lib\site-packages (from transformers->-r requirements.txt (line 20)) (2024.5.15)
Requirement already satisfied: requests in e:\programme\python\python310\lib\site-packages (from transformers->-r requirements.txt (line 20)) (2.32.3)
Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in e:\programme\python\python310\lib\site-packages (from transformers->-r requirements.txt (line 20)) (0.13.3)
WARNING: opencv-python-headless 4.7.0.72 does not provide the extra 'ffmpeg'
Requirement already satisfied: smmap<6,>=3.0.1 in e:\programme\python\python310\lib\site-packages (from gitdb<5,>=4.0.1->gitpython->-r requirements.txt (line 5)) (5.0.1)
Requirement already satisfied: fsspec>=2023.5.0 in e:\programme\python\python310\lib\site-packages (from huggingface_hub->timm>=0.4.12->-r requirements.txt (line 18)) (2024.6.1)
Requirement already satisfied: typing-extensions>=3.7.4.3 in e:\programme\python\python310\lib\site-packages (from huggingface_hub->timm>=0.4.12->-r requirements.txt (line 18)) (4.12.2)
Requirement already satisfied: six>=1.5 in e:\programme\python\python310\lib\site-packages (from python-dateutil>=2.7->matplotlib->-r requirements.txt (line 8)) (1.16.0)
Requirement already satisfied: sympy in e:\programme\python\python310\lib\site-packages (from torch>=1.8.0->fairscale>=0.4.4->-r requirements.txt (line 2)) (1.12.1)
Requirement already satisfied: jinja2 in e:\programme\python\python310\lib\site-packages (from torch>=1.8.0->fairscale>=0.4.4->-r requirements.txt (line 2)) (3.1.4)
Requirement already satisfied: attrs>=22.2.0 in e:\programme\python\python310\lib\site-packages (from jsonschema->rembg->-r requirements.txt (line 14)) (23.2.0)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in e:\programme\python\python310\lib\site-packages (from jsonschema->rembg->-r requirements.txt (line 14)) (2023.12.1)
Requirement already satisfied: referencing>=0.28.4 in e:\programme\python\python310\lib\site-packages (from jsonschema->rembg->-r requirements.txt (line 14)) (0.35.1)
Requirement already satisfied: rpds-py>=0.7.1 in e:\programme\python\python310\lib\site-packages (from jsonschema->rembg->-r requirements.txt (line 14)) (0.18.1)
Requirement already satisfied: coloredlogs in e:\programme\python\python310\lib\site-packages (from onnxruntime->rembg->-r requirements.txt (line 14)) (15.0.1)
Requirement already satisfied: flatbuffers in e:\programme\python\python310\lib\site-packages (from onnxruntime->rembg->-r requirements.txt (line 14)) (24.3.25)
Requirement already satisfied: protobuf in e:\programme\python\python310\lib\site-packages (from onnxruntime->rembg->-r requirements.txt (line 14)) (4.25.3)
Requirement already satisfied: platformdirs>=2.5.0 in e:\programme\python\python310\lib\site-packages (from pooch->rembg->-r requirements.txt (line 14)) (4.2.2)
Requirement already satisfied: charset-normalizer<4,>=2 in e:\programme\python\python310\lib\site-packages (from requests->transformers->-r requirements.txt (line 20)) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in e:\programme\python\python310\lib\site-packages (from requests->transformers->-r requirements.txt (line 20)) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in e:\programme\python\python310\lib\site-packages (from requests->transformers->-r requirements.txt (line 20)) (2.2.2)
Requirement already satisfied: certifi>=2017.4.17 in e:\programme\python\python310\lib\site-packages (from requests->transformers->-r requirements.txt (line 20)) (2024.6.2)
Requirement already satisfied: humanfriendly>=9.1 in e:\programme\python\python310\lib\site-packages (from coloredlogs->onnxruntime->rembg->-r requirements.txt (line 14)) (10.0)
Requirement already satisfied: MarkupSafe>=2.0 in e:\programme\python\python310\lib\site-packages (from jinja2->torch>=1.8.0->fairscale>=0.4.4->-r requirements.txt (line 2)) (2.1.5)
Requirement already satisfied: mpmath<1.4.0,>=1.1.0 in e:\programme\python\python310\lib\site-packages (from sympy->torch>=1.8.0->fairscale>=0.4.4->-r requirements.txt (line 2)) (1.3.0)
Requirement already satisfied: pyreadline3 in e:\programme\python\python310\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime->rembg->-r requirements.txt (line 14)) (3.4.1)
Using cached cmake-3.30.2-py3-none-win_amd64.whl (35.6 MB)
Using cached opencv_python_headless-4.7.0.72-cp37-abi3-win_amd64.whl (38.1 MB)
Using cached pilgram-1.2.1-py3-none-any.whl (817 kB)
Using cached rembg-2.0.58-py3-none-any.whl (33 kB)
Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl (2.5 MB)
Using cached pooch-1.8.2-py3-none-any.whl (64 kB)
Using cached PyMatting-1.1.12-py3-none-any.whl (52 kB)
Building wheels for collected packages: img2texture, cstr, ffmpy
  Building wheel for img2texture (pyproject.toml) ... done
  Created wheel for img2texture: filename=img2texture-1.0.6-py3-none-any.whl size=8448 sha256=e3529fa19ff735ece88e762f5c7c1746e00f1aa222c9baf93b73c33b3cbd292c
  Stored in directory: C:\Users\danie\AppData\Local\Temp\pip-ephem-wheel-cache-n2jflge1\wheels\ef\01\9f\2a02c1af22973c81e59b3e7c5a63a57bc73050b9176aa479d2
  Building wheel for cstr (pyproject.toml) ... done
  Created wheel for cstr: filename=cstr-0.1.0-py3-none-any.whl size=2847 sha256=a57a9aa0274a7988534642de7a32540dce53d56ba581865ce8e9630bc8429651
  Stored in directory: C:\Users\danie\AppData\Local\Temp\pip-ephem-wheel-cache-n2jflge1\wheels\89\3f\37\64121630832f72f13d3af8bb065e97483aa6e038bb27dd941a
  Building wheel for ffmpy (pyproject.toml) ... done
  Created wheel for ffmpy: filename=ffmpy-0.3.0-py3-none-any.whl size=5542 sha256=14154346c6248e0b2ef28f6e62ee28faba943a40147dc3f3764f6771ecac7721
  Stored in directory: C:\Users\danie\AppData\Local\Temp\pip-ephem-wheel-cache-n2jflge1\wheels\58\b8\b4\c142357beed3083ac6f76bff1206bb5923ac472c4a740e6921
Successfully built img2texture cstr ffmpy
Installing collected packages: ffmpy, cstr, pillow, pilgram, opencv-python-headless, cmake, pymatting, pooch, img2texture, fairscale, rembg
  Attempting uninstall: ffmpy
    Found existing installation: ffmpy 0.3.2
    Uninstalling ffmpy-0.3.2:
      Successfully uninstalled ffmpy-0.3.2
  Attempting uninstall: pillow
    Found existing installation: pillow 10.3.0
    Uninstalling pillow-10.3.0:
      Successfully uninstalled pillow-10.3.0
  Attempting uninstall: opencv-python-headless
    Found existing installation: opencv-python-headless 4.10.0.84
    Uninstalling opencv-python-headless-4.10.0.84:
      Successfully uninstalled opencv-python-headless-4.10.0.84
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
albucore 0.0.12 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
albumentations 1.4.10 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
blendmodes 2024.1.1 requires Pillow<11,>=10.2.0, but you have pillow 9.5.0 which is incompatible.
Successfully installed cmake-3.30.2 cstr-0.1.0 fairscale-0.4.13 ffmpy-0.3.0 img2texture-1.0.6 opencv-python-headless-4.7.0.72 pilgram-1.2.1 pillow-9.5.0 pooch-1.8.2 pymatting-1.1.12 rembg-2.0.58
```

Again the errors alone:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
albucore 0.0.12 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
albumentations 1.4.10 requires opencv-python-headless>=4.9.0.80, but you have opencv-python-headless 4.7.0.72 which is incompatible.
blendmodes 2024.1.1 requires Pillow<11,>=10.2.0, but you have pillow 9.5.0 which is incompatible.
```

**Comment by WASasquatch** [comment]
Don't use the same venv as other software. Create a new venv for every
project.

**Comment by mattymatty** [comment]
What? That makes no sense. Create a new conda environment outside the comfy environment?

**Comment by WAS-PlaiLabs** [comment]
Stable Diffusion WebUI, or Automatic1111 or variants is a competing system which uses different dependencies than ComfyUI. You don't install them in the same conda/venv/docker.

**Comment by schoenid** [comment]
> Don't use the same venv as other software. Create a new venv for every project.

Using the` Stable Diffusion WebUI` with the `ComfyUI extension`, the same venv is used. This is by 'default' and it is not clear for me, how to change that and why it is not done by default, if it would be possible or required. 
(@WASasquatch )

> Stable Diffusion WebUI, or Automatic1111 or variants is a competing system which uses different dependencies than ComfyUI. You don't install them in the same conda/venv/docker.

As I know, I can't do anything, that influences whether to use or not to use the same dependencies, while installing the ComfyUI Extension in Stable Diffusion WebUI or Automatic1111. 
I'm installing it by using the WebUI Extension installer and I don't have any idea, whats going on behind. I only see the results. 
The same is for the ComfyUI Nodes and Models. I'm just using the ComfyUI Manager. 
(@WAS-PlaiLabs )

If the WebUI crashes and won't come up after an installation of such a node or model, this tells me, that both are using the same environment. A possibility to recover is, as I have learned now, to delete the venv folder inside the base WebUI folder, to remove the last installed models or nodes manually and then start the WebUI again. This takes some time ...
I think, I should do a backup of the `venv` and `ComfyUI\custom nodes` folders in the future before any node installation, to prevent from such disasters.

**Comment by WASasquatch** [comment]
Ah you are using the extension. That is a pretty old extension which hasn't had an update in 7mo. The thing about this is getting ConfyUI and A1111 together isn't the biggest problem, it's custom nodes using whatever versions of things. Often the newest which is a pain cause they often aren't using the new features in it. 

I personally would suggest trying something like CushyStudio over using A1111. I've never had a good time with it and it's scripts let alone supporting ConfyUI's custom node ecosystem.

**Comment by schoenid** [comment]
Well, thank you, @WASasquatch .

I do understand now, that the problem lies really on the extensions side, respectevely in conflicts within the used environments of A1111 and ConfyUI. There is no proper separation.

I've tried a parallel installation of the standalone ComfyUI portable now, which works really fine and without such problems.
Thanks to `extra_model_paths.yaml` I can use all models on both systems.

I've installed all the former problematic nodes and models **all at once** and had no problems !!!
(maybe except the ReActor node, but this is a different story with insightface, which has to be installed manually.)
